File: android/src/main/java/expo/modules/infinitemedia/Log.kt
package expo.modules.infinitemedia

internal object Log {
  private const val TAG = "InfiniteMedia"

  /** 0 none, 1 error, 2 warn, 3 debug. */
  @Volatile var level = 2

  fun setLevel(name: String?) {
    level = when (name) {
      "none" -> 0
      "error" -> 1
      "debug" -> 3
      else -> 2
    }
  }

  fun d(msg: String) { if (level >= 3) android.util.Log.d(TAG, msg) }
  fun w(msg: String, t: Throwable? = null) { if (level >= 2) android.util.Log.w(TAG, msg, t) }
  fun e(msg: String, t: Throwable? = null) { if (level >= 1) android.util.Log.e(TAG, msg, t) }
}

File: ios/Support/Log.swift
import os

enum LogLevel: Int {
  case none = 0, error, warn, debug

  init(name: String?) {
    switch name {
    case "none": self = .none
    case "error": self = .error
    case "debug": self = .debug
    default: self = .warn
    }
  }
}

enum Log {
  static var level: LogLevel = .warn
  private static let logger = Logger(subsystem: "expo.modules.infinitemedia", category: "InfiniteMedia")

  static func error(_ message: @autoclosure () -> String) {
    if level.rawValue >= LogLevel.error.rawValue { let m = message(); logger.error("\(m, privacy: .public)") }
  }

  static func warn(_ message: @autoclosure () -> String) {
    if level.rawValue >= LogLevel.warn.rawValue { let m = message(); logger.warning("\(m, privacy: .public)") }
  }

  static func debug(_ message: @autoclosure () -> String) {
    if level.rawValue >= LogLevel.debug.rawValue { let m = message(); logger.debug("\(m, privacy: .public)") }
  }
}
